### ISSUE TITLE
Search path for settings files: unclear behavior in the documentation

### ISSUE THREAD

**Issue by antoine-gallix** [dynaconf/dynaconf#832]
In the documentation, at `setting files > setting file location` it is mentioned that `Dynaconf will search files specified in settings_file option starting the search tree on the current working dir (the directory where your program is located).`

I find it confusing because in my mind, the current working directory is the location of my shell  when I run the program, and the directory where my program is located can be a different one. From today's debugging session it appears that the start of the search is the current working directory, but NOT the directory where the program is located.

In my case I have the following file structure:

```
home/
├── application # version controlled application
│   ├── config-defaults.toml # version controlled default settings
│   └── config.py # the config is build here
└── config.toml # local configuration, with secrets
```

In `config.py` the config is build with
```
import dynaconf

config = dynaconf.Dynaconf(
    settings_file=["config-defaults.toml", "config.toml"],
    merge_enabled=True
)
```
If I load the config from the `application` directory, the default settings are loaded, and can be overridden by the system-specific settings. But if I call it from a higher working directory, like `home`, then the default configuration is not loaded. Should I change the documentation to clarify this?

**Comment by rochacbruno** [comment]
We want to add the flexibility to change the file loading strategy, so in 4.0 there will be a way to set

```python
file_finder=Default|Pwd|Absolute|Custom
```

**Comment by pedro-psb** [comment]
Hopefully closed by #894

**Comment by antoine-gallix** [comment]
Those docs changes make things clearer. thanks.
